woocommerce/woocommerce-ios
# Product Sharing: Add unit tests for ShareProductCoordinator

Part of #9867 


## Description

This PR cleans up the code for `ShareProductCoordinator` and adds unit tests for it. I wanted to test the dismissal event of the AI sheet but it's only tracked when the sheet is dismissed interactively, not programmatically, so it's not possible unless we change the implementation.

## Testing instructions

Some refactoring was done to the `FirstProductCreatedView` and view model, so please test again to make sure that all the product-sharing features still work properly.

## Screenshots

N/A

---
